packages/browser/src/posthog-core.ts

```diff
@@ -1688,6 +1689,73 @@ export class PostHog {
         this.featureFlags.reloadFeatureFlags()
     }
 
+    /**
+     * Manually update feature flag values without making a network request.
+     *
+     * This is useful when you have feature flag values from an external source
+     * (e.g., server-side evaluation, edge middleware) and want to inject them
+     * into the client SDK.
+     *
+     * {@label Feature flags}
+     *
+     * @example
+     * ```js
+     * // Replace all flags with server-evaluated values
+     * posthog.updateFlags({
+     *   'my-flag': true,
+     *   'my-experiment': 'variant-a'
+     * })
+     *
+     * // Merge with existing flags (update only specified flags)
+     * posthog.updateFlags(
+     *   { 'my-flag': true },
+     *   undefined,
+     *   { merge: true }
+     * )
+     *
+     * // With payloads
+     * posthog.updateFlags(
+     *   { 'my-flag': true },
+     *   { 'my-flag': { some: 'data' } }
+     * )
+     * ```
+     *
+     * @param flags - An object mapping flag keys to their values (boolean or string variant)
+     * @param payloads - Optional object mapping flag keys to their JSON payloads
+     * @param options - Optional settings. Use `{ merge: true }` to merge with existing flags instead of replacing.
+     * @public
+     */
+    updateFlags(
+        flags: Record<string, boolean | string>,
+        payloads?: Record<string, JsonType>,
+        options?: { merge?: boolean }
+    ): void {
+        // If merging, combine with existing flags
+        const existingFlags = options?.merge ? this.featureFlags.getFlagVariants() : {}
+        const existingPayloads = options?.merge ? this.featureFlags.getFlagPayloads() : {}
+        const mergedFlags = { ...existingFlags, ...flags }
+        const mergedPayloads = { ...existingPayloads, ...payloads }
```

Comment:
Minor nit: If `merge: false` then these aren't technically merged flags. Perhaps `finalFlags` and `finalPayloads` would be more accurate names? Regardless, I'm fine with the name as-is too.